build/publish-javadoc.sh

```diff
@@ -1,48 +1,46 @@
 #!/bin/bash
 
-# Avoid publishing javadocs for a PR build
-if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_BRANCH" ]; then
-
-    printf "\n>>>>> Publishing javadoc for release build: repo=%s branch=%s build_num=%s job_num=%s\n" ${TRAVIS_REPO_SLUG} ${TRAVIS_BRANCH} ${TRAVIS_BUILD_NUMBER} ${TRAVIS_JOB_NUMBER} 
-
-    printf "\n>>>>> Cloning repository's gh-pages branch into directory 'gh-pages'"
-    git clone --branch=gh-pages https://${GH_TOKEN}@github.com/IBM/platform-services-java-sdk.git gh-pages
-
-    printf "\n>>>>> Finished cloning...\n"
-
-    
-    pushd gh-pages
-    
-    # Create a new directory for this branch/tag and copy the aggregated javadocs there.
-    printf "\n>>>>> Copying aggregated javadocs to new tagged-release directory: %s\n" ${TRAVIS_BRANCH}
-    rm -rf docs/${TRAVIS_BRANCH}
-    mkdir -p docs/${TRAVIS_BRANCH}
-    cp -rf ../target/site/apidocs/* docs/${TRAVIS_BRANCH}
-
-    printf "\n>>>>> Generating gh-pages index.html...\n"
-    ../build/generate-index-html.sh > index.html
-
-    # Update the 'latest' symlink to point to this branch if it's a tagged release.
-    if [ -n "$TRAVIS_TAG" ]; then
-	pushd docs
-	rm latest
-	ln -s ./${TRAVIS_TAG} latest
-	printf "\n>>>>> Updated 'docs/latest' symlink:\n"
-	ls -l latest
-	popd
-    fi
-
-    printf "\n>>>>> Committing new javadoc...\n"
-    git add -f .
-    git commit -m "Javadoc for release ${TRAVIS_TAG} (${TRAVIS_COMMIT})"
-    git push -f origin gh-pages
-
-    popd
+# Store GIT properties in vars
```

Comment:
Shouldn't we use an if/else block to only publish docs for commits merged into master?

Reply:
Currently, the Jenkins will publish a javadoc when it is a tagged public release, see the related CI change in the `sdks` repo.
Here do you suggest to change this process to publish a javadoc per master merge?

Reply:
No that's fine to publish on tagged releases.